netmaster/master/api.go

```diff
@@ -190,6 +191,40 @@ func AllocAddressHandler(w http.ResponseWriter, r *http.Request, vars map[string
 	}
 
 	if networkID == "" {
+		if GetClusterMode() == "docker-compose" {
+			// If the network was created using docker command,
+			// we get allocReq before the network is create. Here,
+			// we return the first IP in the subnet as gateway IP.
+			// Subsequent createNetwork request will come with
+			// a subnet and gateway which will allocate the gateway IP
+			var ipAddress string
+			var subnetLen uint
+			subnetIP := strings.Split(allocReq.AddressPool, "/")[0]
+			subnetLen = func() uint { p, _ := strconv.Atoi(strings.Split(allocReq.AddressPool, "/")[1]); return uint(p) }()
```

Comment:
error check here too.  Might want to write it with regular formatting instead of as a one-liner